core/src/validator.rs

```diff
@@ -746,6 +751,52 @@ fn active_vote_account_exists_in_bank(bank: &Arc<Bank>, vote_account: &Pubkey) -
     false
 }
 
+fn compute_poh_sleep(
+    poh_sleep_ns: Option<u64>,
+    genesis_config: &GenesisConfig,
+    bank: &Bank,
+) -> u64 {
+    match poh_sleep_ns {
+        Some(sleep_ns) => sleep_ns,
+        None => match genesis_config.hashes_per_tick() {
+            Some(hashes_per_tick) => {
+                let ticks_per_slot = bank.ticks_per_slot();
+                let hashes_per_slot = hashes_per_tick * ticks_per_slot;
+                let hash_samples = 1_000_000;
+                let hash_time_ns = compute_hash_time_ns(hash_samples);
+                let my_ns_per_slot = (hash_time_ns * hashes_per_slot) / hash_samples;
+                debug!(
+                    "computed: ns_per_slot: {} hash_time_ns: {}",
+                    my_ns_per_slot, hash_time_ns
+                );
+                let iterations = hashes_per_slot / crate::poh_service::NUM_HASHES_PER_BATCH;
+                let target_ns_per_slot = genesis_config.ns_per_slot() as u64;
+                debug!(
+                    "cluster ns_per_hash: {}ns ns_per_slot: {}",
+                    target_ns_per_slot / hashes_per_slot,
+                    target_ns_per_slot
+                );
+                if my_ns_per_slot < target_ns_per_slot {
+                    let extra_ns = target_ns_per_slot - my_ns_per_slot;
+                    let sleep_ns = extra_ns / iterations;
+                    info!(
+                        "Needed {}ns extra total. Calculated {} ns extra sleep",
+                        extra_ns, sleep_ns
+                    );
+                    sleep_ns
+                } else {
+                    warn!(
+                        "Poh is slower than cluster target tick rate! mine: {} cluster: {}",
+                        my_ns_per_slot, target_ns_per_slot,
+                    );
```

Comment:
A new operator doesn't want to waste time configuring a slow machine that will fall-over in production.

It might be helpful & beneficial for the operator to "fail fast" and terminate the process after writing the log message. When the valdiator terminates, they can check the log and quicky see what went wrong. I suppose it will be helpful to create an `--allow-slow-poh` flag to over-ride this behavior if they desire.

A simpler alternative might be to write the warning message in all-caps to get the operator's attention!

Reply:
Ok. I added `panic` and an argument to disable.